src/Lookup/MultiplexingEntityLookup.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\EntityDocument;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikimedia\Assert\Assert;
+use Wikimedia\Assert\ParameterAssertionException;
+use Wikimedia\Assert\ParameterElementTypeException;
+
+/**
+ * Wrapper around repository-specific EntityLookups
+ * picking up the right lookup object for the particular input.
+ *
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ */
+class MultiplexingEntityLookup implements EntityLookup {
```

Comment:
Since I was fuzzy on what exactly multiplexing means, I had to look it up (not good). This service is not combining any of its input, so I think the name is misleading.

We have similar services where we used something like `DispatchingFooService` or `DelegatingFooService`. Would be nice to include a hint that this is done based on repository name.